cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,128 @@
-                    in this case only one.
-            * Verify that appliance's database contains such entries in table ``event_streams``
-        Metadata:
-            test_flag: rest
-        """
-        event = {
-            "event_type": "BadUserNameSessionEvent",
-            "event_message": "Cannot login user@test.domain {}".format(from_detail)
-        }
-        rest_vm = appliance.rest_api.collections.vms.get(name=vm)
-        if from_detail:
-            responses = [rest_vm.action.add_event(**event)]
-        else:
-            responses = appliance.rest_api.collections.vms.action.add_event(rest_vm, **event)
-        assert appliance.rest_api.response.status_code == 200
-        for response in responses:
-            assert response["success"] is True, "Could not add event"
-
-        events = appliance.db.client["event_streams"]
-        events_list = list(appliance.db.client.session.query(events).filter(
-            events.vm_name == vm,
-            events.message == event["event_message"],
-            events.event_type == event["event_type"],
-        ))
-        assert events_list, "Could not find the event in the database"
-
-    @pytest.mark.parametrize("from_detail", [True, False], ids=["from_detail", "from_collection"])
-    def test_vm_add_lifecycle_event(self, vm, from_detail, appliance):
-        """Test that checks whether adding a lifecycle event using the REST API works.
-        Prerequisities:
-            * A VM
-        Steps:
-            * Find the VM's id
-            * Prepare the lifecycle event data (``status``, ``message``, ``event``)
-            * Call either:
-                * POST /api/vms/<id> (method ``add_lifecycle_event``) <- the lifecycle data
-                * POST /api/vms (method ``add_lifecycle_event``) <- the lifecycle data
-                    and resources field specifying list of dicts containing hrefs to the VMs,
-                    in this case only one.
-            * Verify that appliance's database contains such entries in table ``lifecycle_events``
-        Metadata:
-            test_flag: rest
-        """
-        rest_vm = appliance.rest_api.collections.vms.get(name=vm)
-        event = dict(
-            status=fauxfactory.gen_alphanumeric(),
-            message=fauxfactory.gen_alphanumeric(),
-            event=fauxfactory.gen_alphanumeric(),
-        )
-        if from_detail:
-            responses = [rest_vm.action.add_lifecycle_event(**event)]
+    def __init__(self, vm, event):
+        self.vm = vm
+        self.action = event
+
+        if self.action == 'create':
+            self.tl_event = 'VmDeployedEvent'
+            self.tl_category = 'Creation/Addition'
+            self.db_event = ''
+            self.emit_cmd = self._setup_vm()
+        elif self.action in {'start', 'resume'}:
+            self.tl_event = 'VmPoweredOnEvent'
+            self.tl_category = 'Power Activity'
+            self.db_event = ''
+            self.emit_cmd = self._power_on()
```

Comment:
REQUIRED you call all these functions in init. the plan was to call them by emit method.